From b252133d765245a297e446c805822601e5a4924b Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 27 Sep 2013 13:50:20 +0000
Subject: main: Fix possible assertion on startup

In some cases, Totem was focused on startup by the time we
handled command-line arguments. This triggered an assertion
when trying to show the fullscreen popups because we weren't
fullscreened yet.

https://bugzilla.gnome.org/show_bug.cgi?id=708647
---
diff --git a/src/totem-fullscreen.c b/src/totem-fullscreen.c
index fac1b05..69e1fd7 100644
--- a/src/totem-fullscreen.c
+++ b/src/totem-fullscreen.c
@@ -307,7 +307,8 @@ totem_fullscreen_show_popups (TotemFullscreen *fs, gboolean show_cursor)
 {
 	GtkWidget *item;
 
-	g_assert (fs->priv->is_fullscreen != FALSE);
+	if (!fs->priv->is_fullscreen)
+		return;
 
 	if (fs->priv->popup_in_progress != FALSE)
 		return;
--
cgit v0.9.2
